controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -878,6 +878,16 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 		return result
 	}
 
+	dirty, err = r.matchProfile(info)
+	if err != nil {
+		// FIXME(zaneb): This error requires a Spec change to fix, so we
+		// shouldn't treat it as transient
+		return actionError{errors.Wrap(err, "failed to match profile")}
```

Comment:
Now that we are doing this in a state where setting ProvisioningError makes some sense, can we fix this?